src/olympia/abuse/models.py

```diff
@@ -235,6 +231,20 @@ def notify_reporters(self, action_helper):
             is_appeal=True,
         )
 
+    def handle_job_recreated(self, new_job_id):
+        new_job, _ = CinderJob.objects.update_or_create(
+            job_id=new_job_id,
+            defaults={
+                'resolvable_in_reviewer_tools': True,
+                'target_addon': self.target_addon,
+            },
+        )
+        # Update our fks to connected objects
+        AbuseReport.objects.filter(cinder_job=self).update(cinder_job=new_job)
+        AbuseReport.objects.filter(appellant_job=self).update(appellant_job=new_job)
```

Comment:
I think this fk is unnecessary now we have the `CinderAppeal` model, but existing code relies on it so I'll remove in a follow-up.

Reply:
https://github.com/mozilla/addons/issues/14984